src/main/scala/sigmastate/eval/RuntimeCosting.scala

```diff
@@ -648,9 +648,12 @@ trait RuntimeCosting extends CostingRules with DataCosting with Slicing { IR: Ev
 //        RCostedColl(vals, costs, sizes, xs.valuesCost)
 
 
-      case CostedM.cost(Def(CCostedCollCtor(_, costs, _, accCost))) => opCost(Seq(accCost), costs.sum(intPlusMonoid))
-      case CostedM.cost(Def(CCostedOptionCtor(_, costOpt, _, accCost))) => opCost(Seq(accCost), costOpt.getOrElse(Thunk(0)))
-      case CostedM.cost(Def(CCostedPairCtor(l, r, accCost))) => opCost(Seq(accCost), l.cost + r.cost)
+      case CostedM.cost(Def(CCostedCollCtor(values, costs, _, accCost))) =>
+        opCost(values, Seq(accCost), costs.sum(intPlusMonoid))
+      case CostedM.cost(Def(CCostedOptionCtor(v, costOpt, _, accCost))) =>
+        opCost(v, Seq(accCost), costOpt.getOrElse(Thunk(0)))
+      case CostedM.cost(Def(CCostedPairCtor(l, r, accCost))) =>
+        opCost(l.value, Seq(accCost), l.cost + r.cost)
```

Comment:
@greenhat for the `CostedPairCtor` case try this `opCost(Pair(l.value, r.value), Seq(accCost), l.cost + r.cost)`
This Pair is more accurate value which is costed

Reply:
Fixed.